doc/contributing/style_guides/markdown_usage_style.md

```diff
@@ -37,277 +39,131 @@ No style guide is perfect.
-
-
-```
-
-## Hardware Interfaces
-
-
-```
-
-The organization of the design details section is done to suit the module.
-
-```
-
-## Design Details
-
-Details of the design.
-
-### Many third level headings
-```
-There are probably waveforms embedded here:
-
-```
-
-{{</* wavejson */>}}
-{
-  signal: [
-    { name: 'Clock',        wave: 'p............' },
-  ]
-}
-{{</* /wavejson */>}}
-
-```
-
-The final major section is the software user guide and describes using the IP and notes on writing device drivers.
-Code fragments are encouraged.
-
-_This section is primarily for software engineers, but it is expected that the code fragments are used by verification engineers._
-
-```
-
-# Programmers Guide
 
-```
+## Filename
 
-One important thing here is to show the order of initialization that has been tested in the verification environment.
-In most cases other orders will work, and may be needed by the software environment, but it can be helpful in tracking down bugs for the validated sequence to be described!
+The Markdown file names should be in snake case and should use the `.md` file extension.
 
-```
-## Initialization
-```
-```c
+## Extensions
 
- if (...) {
-   a = ...
- }
-```
+One must only use [CommonMark][] with the exception of the following extensions:
+- [GFM Tables](https://github.github.com/gfm/#tables-extension-)
+- [GFM List Items](https://github.github.com/gfm/#task-list-items-extension-)
+- [Inline maths and maths blocks](#maths)
 
-Other sections cover different use cases and example code fragments.
+## Line length and Whitespace
 
-```
+Files must have exactly one sentence per line, with no line breaks in the middle of a sentence.
+This way change reviews will highlight only those sentences which are modified.
+Though the long line lengths make the files slightly less convenient to read from the command-line, this greatly simplifies the review process.
+When reviewing Markdown changes, every altered sentence will be included in its entirety in the file diff.
 
-## Use case A (eg Transmission)
+Markdown files must contain no trailing white space.
+As a result, two or more spaces must not be used for [hard line breaks][].
+Instead, a double new line is recommended.
+[Backslash hard breaks][] can also be used.
 
-## Use case B (eg Reception)
+## HTML
 
-```
+Do not use inline HTML within markdown documents.
+Although it offers a lot of flexibility, this flexibility is hard to handle.
+- It is possible to break the website
+- Markdown renderers often support different subsets of HTML.
+- The use of HTML restricts the ease at which documentation could be converted to other forms in the future, such as PDF.
 
-It is important to include a discussion of error conditions.
+Reaching for HTML may suggest that:
+- You need to split the concepts into more smaller parts that can be expressed in markdown syntax.
+- The concept would be better explained in an svg diagram.
```

Comment:
Do we make any exception for tables, or is restructuring them going to be enforced?

I tend to think splitting is the right answer, but in some cases may not be possible